src/test/java/net/kaczmarzyk/e2e/annotated/AnnotatedSpecInterfaceAndE2eTest.java

```diff
@@ -0,0 +1,264 @@
+package net.kaczmarzyk.e2e.annotated;
+
+import net.kaczmarzyk.E2eTestBase;
+import net.kaczmarzyk.spring.data.jpa.Customer;
+import net.kaczmarzyk.spring.data.jpa.CustomerRepository;
+import net.kaczmarzyk.spring.data.jpa.domain.Equal;
+import net.kaczmarzyk.spring.data.jpa.domain.In;
+import net.kaczmarzyk.spring.data.jpa.domain.Like;
+import net.kaczmarzyk.spring.data.jpa.domain.LikeIgnoreCase;
+import net.kaczmarzyk.spring.data.jpa.web.annotation.And;
+import net.kaczmarzyk.spring.data.jpa.web.annotation.Spec;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+/**
+ * Test cases:
+ *  TC-1. interface with @And spec
+ *  TC-2. interface with @And spec extended by param spec
+ *  TC-3. interface without any spec extended by param spec
+ *  TC-4. interface without any spec extended by interface with @And spec
+ *  TC-5. interface with @And spec extended by other interface with @And spec
+ *  TC-6. interface with @And spec extended by other interface with @And spec and param spec
+ *  TC-7. interface with @And spec extended by interface with @And spec and extended by interface with @And spec extended by another interface with @And spec
+ *  TC-8. interface with @And spec extended by interface with @And spec and extended by interface with @And spec extended by another interface with @And spec and extended by param spec
+ */
+public class AnnotatedSpecInterfaceAndE2eTest extends E2eTestBase {
+
+	// TC-1. interface with @And spec
+	@And({
+			@Spec(params = "lastName", path = "lastName", spec = Equal.class)
+	})
+	private static interface LastNameAndSpec extends Specification<Customer> {
+	}
+
+	// TC-3. interface without any spec extended by param spec
+	private static interface InterfaceWithoutAnySpec extends Specification<Customer> {}
+
+	// TC-4. interface without any spec extended by interface with @And spec
```

Comment:
I think it's "extending" not "extended", right? Revisit all other cases and correct if needed